pkg/provider/vault/vault.go

```diff
@@ -528,6 +535,14 @@ func (v *client) secretKeyRef(ctx context.Context, secretRef *esmeta.SecretKeySe
 	return valueStr, nil
 }
 
+// checkToken does a lookup and checks if the provided token exists.
+func checkToken(ctx context.Context, vStore *client) error {
+	// https://www.vaultproject.io/api-docs/auth/token#lookup-a-token-self
+	req := vStore.client.NewRequest("GET", "/v1/auth/token/lookup-self")
```

Comment:
As discussed with @gusfcarvalho, we want to have a dedicated `ping` or `healthcheck` method on the interface so we can call that on purpose from the SecretStore controller. This isn't necessarily needed from the `ExternalSecret` controller.

Reply:
https://github.com/external-secrets/external-secrets/issues/467

We have this old issue about the secretstore controller. At the time we were thinking only of status conditions, we can maybe update it and think about closing it or getting a follow up issue